src/sql/workbench/contrib/notebook/browser/outputs/notebookMarkdown.ts

```diff
@@ -98,7 +99,14 @@ export class NotebookMarkdownRenderer {
 			}
 			let attributes: string[] = [];
 			if (href) {
-				attributes.push(`src="${href}"`);
+				// VS Code blocks loading directly from the file protocol - we have to transform it to a vscode-file URI
+				// first. Since the href here can be either a file path, an HTTP/S link or embedded data though we first
+				// check if it's any of the others and if so then don't need to do anything.
+				let uri = URI.parse(href);
+				if (!(uri.scheme === 'https' || uri.scheme === 'http' || uri.scheme === 'data')) {
```

Comment:
This isn't my favorite thing to do but I couldn't think of a better way to determine if a random string is a file path or something else....open to any other suggestions though!

Reply:
What happens for relative paths that are on-disk? `"./foo.png"` Does the scheme get set properly here?

Reply:
Oh never mind, I see now, this is only for non-file schemes. Carry on 😄